### PR TITLE
Repripritize reroute requests

### PR DESCRIPTION
**Pull request by rozdy**
Flows with COST_AND_AVAILABLE_BANDWIDTH path computation strategy should be rerouted from low to high bandwidth

### CODE REVIEW

File: src-java/reroute-topology/reroute-storm-topology/src/main/java/org/openkilda/wfm/topology/reroute/service/RerouteQueueService.java [hunk 1]
```diff
@@ -228,15 +230,31 @@ public int compare(FlowThrottlingData throttlingDataA, FlowThrottlingData thrott
...
+            PathComputationStrategy pathComputationStrategyA = throttlingDataA.getPathComputationStrategy();
+            PathComputationStrategy pathComputationStrategyB = throttlingDataB.getPathComputationStrategy();
 
-            if (priorityA == priorityB && (timeCreateA != null || timeCreateB != null)) {
-                if (timeCreateA == null) {
-                    return -1;
+            if (priorityA == priorityB) {
+                if (pathComputationStrategyA == COST_AND_AVAILABLE_BANDWIDTH
+                        && pathComputationStrategyB == COST_AND_AVAILABLE_BANDWIDTH) {
+                    return Long.compare(throttlingDataB.getBandwidth(), throttlingDataA.getBandwidth());
```
Comment (dpoltavets) [inline src-java/reroute-topology/reroute-storm-topology/src/main/java/org/openkilda/wfm/topology/reroute/service/RerouteQueueService.java:239]: What if we have the same bandwidths? Shouldn't we then compare by creation time?

Comment (rozdy) [reply src-java/reroute-topology/reroute-storm-topology/src/main/java/org/openkilda/wfm/topology/reroute/service/RerouteQueueService.java]: Fixed

File: src-java/reroute-topology/reroute-storm-topology/src/main/java/org/openkilda/wfm/topology/reroute/service/RerouteQueueService.java [hunk 2]
```diff
@@ -228,15 +230,35 @@ public int compare(FlowThrottlingData throttlingDataA, FlowThrottlingData thrott
             int priorityB = throttlingDataB.getPriority() == null ? defaultFlowPriority : throttlingDataB.getPriority();
             Instant timeCreateA = throttlingDataA.getTimeCreate();
             Instant timeCreateB = throttlingDataB.getTimeCreate();
+            PathComputationStrategy pathComputationStrategyA = throttlingDataA.getPathComputationStrategy();
```
Comment (surabujin) [inline src-java/reroute-topology/reroute-storm-topology/src/main/java/org/openkilda/wfm/topology/reroute/service/RerouteQueueService.java:233]: As for me it should be separate comparator joined to existing via `thenComparing`.

Comment (rozdy) [reply src-java/reroute-topology/reroute-storm-topology/src/main/java/org/openkilda/wfm/topology/reroute/service/RerouteQueueService.java]: Reworked

**Pull request merged**